### PR TITLE
Update fluentd configuration files to v1.7

### PR DESCRIPTION
**Pull request by MiroslavVavra**
Related to update of fury-images/fluentd-elasticsearch to v1.7.2

### CODE REVIEW

**Comment by lnovara** [comment]
LGTM 👍

**Comment by lnovara** [comment]
@MiroslavVavra How did you test that the log parsing was working as expected?

**Comment by phisco** [comment]
LGTM, we'll wait for https://github.com/sighupio/fury-images/pull/20 to be merged before merging this one, so that we can also set the correct default image in the fluentd DaemonSet.

**Comment by phisco** [comment]
@MiroslavVavra can you update the deployment to correctly use `quay.io/sighup/fluentd-elasticsearch:v1.7.2-debian-1.0`?

**Comment by phisco** [comment]
@MiroslavVavra as said in the previous comment you have to use `quay.io/sighup/fluentd-elasticsearch:v1.7.2-debian-1.0` and not `sighup/fluentd-elasticsearch:1.7.2`, can you fix it?
I've been testing the new configuration and it looks like the removal of some code from the entrypoint has broken the trick we were using to pass the config files via configmap, could you please fix that?

**Comment by MiroslavVavra** [comment]
@phisco The image version for the deployment was updated to the right tag.  The ConfigMap will follow.

**Comment by MiroslavVavra** [comment]
> @MiroslavVavra How did you test that the log parsing was working as expected?

@lnovara the log parsing was tested on cluster by @phisco I will double check the log parsing with him.

File: katalog/fluentd/fluent.conf [hunk 1]
```diff
@@ -17,15 +20,20 @@
...
    @id out_es
-   log_level info
+   @log_level info
    include_tag_key true
    host "#{ENV['FLUENT_ELASTICSEARCH_HOST']}"
    port "#{ENV['FLUENT_ELASTICSEARCH_PORT']}"
+   path "#{ENV['FLUENT_ELASTICSEARCH_PATH']}"
    scheme "#{ENV['FLUENT_ELASTICSEARCH_SCHEME'] || 'http'}"
    ssl_verify "#{ENV['FLUENT_ELASTICSEARCH_SSL_VERIFY'] || 'true'}"
+   ssl_version "#{ENV['FLUENT_ELASTICSEARCH_SSL_VERSION'] || 'TLSv1'}"
```
Comment (phisco) [inline katalog/fluentd/fluent.conf]: please set `TLSv1_2`

File: katalog/fluentd/fluent.conf [hunk 2]
```diff
@@ -38,32 +46,3 @@
...
-# input plugin that collects metrics for output plugin
-<source>
-  @type prometheus_output_monitor
-  <labels>
-    host ${hostname}
-  </labels>
-</source>
-
-# input plugin that collects metrics for in_tail plugin
-<source>
```
Comment (phisco) [inline katalog/fluentd/fluent.conf]: I see some of the prometheus sources have been removed. Which where the reasons for their existence before (mostly a question for @lnovara ) and which are the reasons that brought to their removal ( @MiroslavVavra )

Comment (lnovara) [reply katalog/fluentd/fluent.conf]: @phisco I've never had a deep look at fluentd metrics but looking at https://github.com/fluent/fluent-plugin-prometheus, `prometheus_output_monitor` should be a superset of `prometheus_monitor` while `prometheus_tail_monitor` doesn't hold valuable metrics (at least from my POV).

If you want, I can deploy both and have a look at the precise differences.

In Fury, we are not relaying on any particular metric to be available as we just alert on the absence of the `up(job="fluentd")` metric.

**Comment by angelbarrera92** [comment]
Is this PR obsolete?

**Comment by lnovara** [comment]
@angelbarrera92 I don't think it's obsolete, I think that we can proceed with the merge and address the two open points shortly after the merge.

What do you think?

cc @phisco @MiroslavVavra

File: katalog/fluentd/deploy.yml [hunk 3]
```diff
@@ -19,7 +19,7 @@ spec:
         effect: NoSchedule
       containers:
       - name: fluentd
-        image: sighup/fluentd-elasticsearch:1.2.8
+        image: sighup/fluentd-elasticsearch:v1.7.2-debian-1.0
```
Comment (lnovara) [inline katalog/fluentd/deploy.yml]: The image does not exist. The right one is: `quay.io/sighup/fluentd-elasticsearch:v1.7.2-debian-1.0`.

**Comment by lnovara** [comment]
I had a closer look at this PR and tried to deploy it on a test cluster. I don't think that this PR can be merged as-is.

**Comment by lnovara** [comment]
@MiroslavVavra can you please have a look?

File: katalog/fluentd/deploy.yml (hunk 3 above)
Comment (MiroslavVavra) [reply katalog/fluentd/deploy.yml]: Yes, I'll fix this.

File: katalog/fluentd/fluent.conf (hunk 1 above)
Comment (MiroslavVavra) [reply katalog/fluentd/fluent.conf]: Setting the TLS version to 1.2

**Comment by angelbarrera92** [comment]
Thanks for this PR. I am working deeply on it!

**Comment by angelbarrera92** [comment]
All this content is now ready to review in #12

**Pull request closed without merging**